### PR TITLE
Add marble drop animation

### PR DESCRIPTION
**Pull request by randomPoison**
I've added a visualization of marbles dropping into a pile in front of the hippo when a player taps. This adds some visual feedback on each tap, which helps the game feel better. I've deliberately implemented this in a server-deterministic way. The server determines the position and color of each marble, and specifies exactly which marble gets removed each time. This ensures that all host displays can render the food pile in the same way.

I've also replaced all usage of "balls" with "marbles".

Progress on #35

### CODE REVIEW

File: www/host.js [hunk 1]
```diff
@@ -76,50 +116,26 @@ socket.onmessage = (event) => {
...
-                    element,
-                    .2,
-                    { left: 0 },
-                    { left: '100px', repeat: 1, yoyo: true, overwrite: 'none' },
-                );
-            } break;
-
-            default: throw new Error('Unrecognized hippo side: ' + hippo.side.side);
-        }
+        let sideName = SIDE_CSS_NAME[hippo.side.side];
```
Comment (wufire) [inline www/host.js:130]: Ah, this is much better.

File: src/game.rs [hunk 2]
```diff
@@ -42,19 +42,10 @@ impl Deserialize for PlayerId {
...
 pub struct PlayerIdGenerator(AtomicUsize);
 
 impl PlayerIdGenerator {
-    /// Creates a new `PlayerIdGenerator`.
-    ///
-    /// Only one `PlayerIdGenerator` should be created in the lifetime of the application. A single
-    /// generator will never create duplicate IDs, but if there are multiple generators will
-    /// produce the same IDs.
-    pub fn new() -> PlayerIdGenerator {
-        PlayerIdGenerator(ATOMIC_USIZE_INIT)
```
Comment (wufire) [inline src/game.rs:55]: How does this differ from the `Default` constructor?

**Review by wufire — COMMENTED** [review]
Looks good! I have one question but nothing blocking.

**Review by wufire — APPROVED** [review]
This one has approve checked, instead of the wrong one (Curse you, `Comment`)

File: src/game.rs (hunk 2 above)
Comment (randomPoison) [reply src/game.rs:55]: Functionally, the `new` constructor (that I removed) does the same thing as the derived `Default::default()` constructor (that I added) is doing. They both initialize the `AtomicUsize` as 0. Generally it's best practice to have both a `Default` impl and a `new` constructor, but in this case we can write less code by automatically deriving the `Default` impl.

**Pull request merged**